### PR TITLE
Typography changes to support light\dark theme

### PR DESCRIPTION
**Pull request by ad9242**
## Summary of the changes
Adds a new theme prop to typography along with appropriate storybook changes and tests
Changes a number of components to override the color of the typography element they use

## Related issue
#1214 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

**Review by GCHQ-Developer-530 — APPROVED** [review]

File: packages/canary-web-components/src/components/ic-tree-item/ic-tree-item.css [hunk 1]
```diff
@@ -130,11 +138,23 @@
...
+    background-color: none !important;
+  }
+
   :host(.ic-tree-item-disabled) .tree-item-content,
   :host(.ic-tree-item-disabled) .arrow-dropdown,
   :host(.ic-tree-item-disabled) ::slotted([slot="icon"]),
   :host(.ic-tree-item-disabled) ::slotted([slot="router-item"]) {
     color: GrayText !important;
+
+    --ic-typography-color: GrayText !important;
```
Comment (gd2910) [inline packages/canary-web-components/src/components/ic-tree-item/ic-tree-item.css:156]: Will this be replaced with a design token in future work?

File: packages/react/src/component-tests/IcTypography/IcTypography.cy.tsx [hunk 2]
```diff
@@ -176,6 +177,18 @@ describe("IcTypography visual regression and a11y tests", () => {
       testThreshold: setThresholdBasedOnEnv(DEFAULT_TEST_THRESHOLD + 0.074),
     });
   });
+
+  it("should test theme prop", () => {
```
Comment (gd2910) [inline packages/react/src/component-tests/IcTypography/IcTypography.cy.tsx:181]:
```suggestion
  it("should render IcTypography with 'inherited', 'light' and 'dark' theme property", () => {
```

File: packages/web-components/src/components/ic-button/ic-button.css
Comment (gd2910) [inline packages/web-components/src/components/ic-button/ic-button.css:1]: Does ic-button not need the color rule changed to --ic-typography-color?

Comment (ad9242) [reply packages/web-components/src/components/ic-button/ic-button.css:1]: have left this for now as button colours will get covered when that component is reviewed\adds dark mode

File: packages/canary-web-components/src/components/ic-tree-item/ic-tree-item.css (hunk 1 above)
Comment (ad9242) [reply packages/canary-web-components/src/components/ic-tree-item/ic-tree-item.css:156]: not these are they are system High Contrast colors that we should be using

**Review by gd2910 — APPROVED** [review]

**Pull request merged**